src/pages/Pool/index.tsx

```diff
@@ -36,45 +37,61 @@ const mobileTabs = [
 ];
 const LIMIT = 8;
 
-export interface PoolExtended extends Pool {
-  pair: PairExtended;
-  hasBalance: boolean;
-}
-
 function PoolPage() {
-  const location = useLocation();
   const screenClass = useScreenClass();
   const [selectedTimeBase, setSelectedTimeBase] = useState(timeBaseOptions[0]);
   const [currentPage, setCurrentPage] = useState(1);
-  const { pairs, findPair } = usePairs();
-  const api = useAPI();
+  const { findPair, getPair } = usePairs();
   const { bookmarks } = usePairBookmark();
-  const [pools, setPools] = useState<PoolExtended[]>([]);
+  const { pools } = usePools();
   const [selectedTabIndex, setSelectedTabIndex] = useState(0);
 
   const [addresses, setAddresses] =
     useState<[string | undefined, string | undefined]>();
   const [selectedPair, setSelectedPair] = useState<PairExtended>();
+  const api = useAPI();
+
+  const balanceQueryResults = useQueries({
```

Comment:
wouldnt it be better to combine `balanceQueryResults` and `balances` like `balances = useQueries(...)..map((item) => item.data);`?

Reply:
I've never thought of writing something like this when using hooks until now lol
I'll try.